### PR TITLE
add typescript typings

### PR DESCRIPTION
**Pull request by dimonnwc3**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Would you mind some test for types like https://github.com/fastify/fastify-cookie/blob/master/test/types.test.ts? We need to check if it compiles fine.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Accepts } from "accepts"
+import * as fastify from "fastify"
+import { Server, IncomingMessage, ServerResponse } from "http"
+
+declare module "fastify" {
+  interface FastifyRequest<HttpRequest> {
```
Comment (Ethan-Arrowood) [inline index.d.ts]: Would something like `interface FastifyRequest<HttpRequest> extends Accepts` work here?

**Comment by dimonnwc3** [comment]
should be updated guys 👍

File: types.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import fastify from "fastify"
+import accepts from "./"
+
+const app = fastify()
+
+app.register(accepts).after(() => {
```
Comment (mcollina) [inline types.test.ts]: there is no need to use `after` here.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Accepts } from "accepts"
+import * as fastify from "fastify"
+import { Server, IncomingMessage, ServerResponse } from "http"
+
+declare module "fastify" {
+  interface FastifyRequest<HttpRequest> extends Accepts {
+    accepts(): Accepts
+  }
+
+  interface FastifyReply<HttpResponse> {
```
Comment (Ethan-Arrowood) [inline index.d.ts]: Can you do the same thing you did to FastifyRequest?

Comment (dimonnwc3) [reply index.d.ts]: I'm afraid it won't work, as FastifyReply has prefixes on all properties, starting with `request`

**Comment by lddubeau** [comment]
Thanks for getting this fix in. It has been merged but is there anything more needed to get this change published? Anything I can help with? The latest released version of the plugin requires installing `@types/fastify-accept`, which is buggy.

**Comment by mcollina** [comment]
Shipped!

**Pull request merged**